spark_client/domain.py

```diff
@@ -0,0 +1,252 @@
+        options: Dict[str, str] = dict()
+
+        if not options_string:
+            return options
+
+        # cleanup quotes
+        line = options_string.strip().replace("'", "").replace('"', "")
+        for arg in line.split("-D")[1:]:
+            kv = arg.split("=")
+            options[kv[0].strip()] = kv[1].strip()
+
+        return options
+
+    @property
+    def options(self) -> Dict[str, Dict]:
+        """Extract properties which are known to be options-like requiring special parsing."""
+        return {
+            k: self._parse_options(v)
+            for k, v in self.props.items()
+            if self._is_property_with_options(k)
+        }
+
+    @staticmethod
+    def _construct_options_string(options: Dict) -> str:
+        result = ""
+        for k in options:
+            v = options[k]
+            result += f" -D{k}={v}"
+        return result
+
+    @classmethod
+    def empty(cls) -> "PropertyFile":
+        """Return an empty property file object."""
+        return PropertyFile(dict())
+
+    def __add__(self, other: "PropertyFile"):
+        return self.union([other])
+
+    def union(self, others: List["PropertyFile"]) -> "PropertyFile":
+        """Merge multiple PropertyFile objects, with right to left priority.
+
+        Args:
+            others: List of Property file to be merged.
+        """
+        all_together = [self] + others
+
+        simple_properties = union(*[prop.props for prop in all_together])
+        merged_options = {
+            k: self._construct_options_string(v)
+            for k, v in union(*[prop.options for prop in all_together]).items()
+        }
+        return PropertyFile(union(*[simple_properties, merged_options]))
+
+
+class Defaults:
+    """Class containing all relevant defaults for the application."""
+
+    def __init__(self, environ: Dict = dict(os.environ)):
+        """Initialize a Defaults class using the value contained in a dictionary
+
+        Args:
+            environ: dictionary representing the environment. Default uses the os.environ key-value pairs.
+        """
+
+        self.environ = environ if environ is not None else {}
+
+    @property
+    def snap_folder(self) -> str:
+        """Return the SNAP folder"""
+        return self.environ["SNAP"]
+
+    @property
+    def static_conf_file(self) -> str:
+        """Return static config properties file packaged with the client snap."""
+        return f"{self.environ.get('SNAP')}/conf/spark-defaults.conf"
+
+    @property
+    def dynamic_conf_file(self) -> str:
+        """Return dynamic config properties file generated during client setup."""
+        return f"{self.environ.get('SNAP_USER_DATA')}/spark-defaults.conf"
+
+    @property
+    def env_conf_file(self) -> Optional[str]:
+        """Return env var provided by user to point to the config properties file with conf overrides."""
+        return self.environ.get("SNAP_SPARK_ENV_CONF")
+
+    @property
+    def snap_temp_dir(self) -> str:
```

Comment:
snap_temp_folder to be consistent in naming?

Reply:
I will address this here!